partiql-ast/src/main/java/org/partiql/ast/v1/AstNode.java

```diff
@@ -0,0 +1,19 @@
+package org.partiql.ast.v1;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collection;
+import java.util.Random;
+
+/**
+ * TODO docs, equals, hashcode
+ */
+public abstract class AstNode {
+    @NotNull
+    public String tag = "Ast-" + String.format("%06x", new Random().nextInt());
+
```

Comment:
I've created a follow-up issue for including source locations. https://github.com/partiql/partiql-lang-kotlin/issues/1608

Reply:
Good point on the source locations -- I'll link the issue.